test/test-asd.js

```diff
@@ -0,0 +1,261 @@
+      test.grabTests(project).then(function () {
+        expect(fs.existsSync(`${path}/test`)).to.be.true;
+        done();
+      });
+    });
+    
+    it('should install product-project tests', function (done) {
+      const project = dummySessionAsd.PROJECT[5];
+      const name = changeCase.paramCase(project.name);
+      const path = `${projectsDirectory}/${name}`;
+      expect(fs.existsSync(`${path}/test`)).to.be.false;
+      projects.ensureProjectsDirectory();
+      fs.mkdirSync(path);
+      fs.writeFileSync(`${path}/package.json`, '{}');
+      test.grabTests(project).then(function () {
+        expect(fs.existsSync(`${path}/test`)).to.be.true;
+        done();
+      });
+    });
+
+    it('should install underpants tests', function (done) {
+      const project = dummySessionAsd.PROJECT[6];
+      const name = changeCase.paramCase(project.name);
+      const path = `${projectsDirectory}/${name}`;
+      expect(fs.existsSync(`${path}/test`)).to.be.false;
+      projects.ensureProjectsDirectory();
+      fs.mkdirSync(path);
+      fs.writeFileSync(`${path}/package.json`, '{}');
+      test.grabTests(project).then(function () {
+        expect(fs.existsSync(`${path}/test`)).to.be.true;
+        done();
+      });
+    });
+
+    it('should install lodown tests', function (done) {
+      const project = dummySessionAsd.PROJECT[7];
+      const name = changeCase.paramCase(project.name);
+      const path = `${projectsDirectory}/${name}`;
+      expect(fs.existsSync(`${path}/test`)).to.be.false;
+      projects.ensureProjectsDirectory();
+      fs.mkdirSync(path);
+      fs.writeFileSync(`${path}/package.json`, '{}');
+      test.grabTests(project).then(function () {
+        expect(fs.existsSync(`${path}/test`)).to.be.true;
+        done();
+      });
+    });
+
+    it('should install lets-get-functional tests', function (done) {
+      const project = dummySessionAsd.PROJECT[8];
+      const name = changeCase.paramCase(project.name);
+      const path = `${projectsDirectory}/${name}`;
+      expect(fs.existsSync(`${path}/test`)).to.be.false;
+      projects.ensureProjectsDirectory();
+      fs.mkdirSync(path);
+      fs.writeFileSync(`${path}/package.json`, '{}');
+      test.grabTests(project).then(function () {
+        expect(fs.existsSync(`${path}/test`)).to.be.true;
+        done();
+      });
+    });
+
+    it('should install first-website tests', function (done) {
+      const project = dummySessionAsd.PROJECT[9];
+      const name = changeCase.paramCase(project.name);
+      const path = `${projectsDirectory}/${name}`;
+      expect(fs.existsSync(`${path}/test`)).to.be.false;
+      projects.ensureProjectsDirectory();
+      fs.mkdirSync(path);
+      fs.writeFileSync(`${path}/package.json`, '{}');
+      test.grabTests(project).then(function () {
+        expect(fs.existsSync(`${path}/test`)).to.be.true;
+        done();
+      });
+    });
+
+    it('should install billypedia tests', function (done) {
```

Comment:
All these tests are doing essentially the same thing, save a few small differences. It would much easier to read and maintain if you generated these tests.
```javascript
dummySessionAsd.PROJECT.forEach((project) => {
  it(`should install ${project.name} tests`, (done) => {
    // Your project test here
  });
});

```
https://mochajs.org/#dynamically-generating-tests